File: examserver/src/main/java/com/exam/services/QuestionService.java
package com.exam.services;

import com.exam.model.exam.Question;
import com.exam.model.exam.Quiz;

import java.util.List;
import java.util.Optional;
import java.util.Set;

public interface QuestionService {
    public  Question addQuestion(Question question);
    public  Question updateQuestion(Question question);
    public  Set<Question> getQuestions();
    public   Question getQuestion(Long quetionId);
    public Set<Question> getQuestionsOfQuiz(Quiz quiz);
    public  void deleteQuestion(Long quesId);
}

File: src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) { }

  public addQuestion(question: any) {
    return this._http.post(`${baseUrl}/question/`, question);
  }

  public getQuestionsOfQuiz(qid: any) {
    return this._http.get(`${baseUrl}/question/quiz/all/${qid}`);
  }

  // Method to get a single question by ID
  public getQuestion(id: number) {
    return this._http.get(`${baseUrl}/question/${id}`);
  }

  // Method to delete a question by ID
  public deleteQuestion(id: number) {
    return this._http.delete(`${baseUrl}/question/${id}`);
  }

  // Method to update a question
  public updateQuestion(id: number, question: any) {
    return this._http.put(`${baseUrl}/question/${id}`, question);
  }
}
